### PR TITLE
WIP - Python3 test fixes

### PR DESCRIPTION
**Pull request by tfmorris**
This uses the Python 3 import cleanup as a base and starts to fix some of the test failures.

Closes #

### CODE REVIEW

**Review by hornc — CHANGES_REQUESTED** [review]
This is great in that it maps out what is needed to get the Python3 tests passing, but simple changes to ensure divisions result in `int`s are going to get held up by changes to requirements.txt.  Moving from web.py 0.39 to 0.40 is going to take quite a bit of planning and discussion to get across the line, this isn't really mergable at its current scale.

**Comment by tfmorris** [comment]
@hornc As a reminder, PRs labelled WIP or Draft are **not** ready for review. I used both on this one to make it extra clear. 

This uses an integration branch with all the `import` fix PRs included. I expect the sequence to be:
* clean up, review, and merge #2888 
* review and merge the other `import` related PRs
* create PRs for the test fixes which are organized by type of fix (e.g. all `iteritems()` fixes) or functional area e.g. `openlibrary.core`

Perhaps you could update the web.py 0.40 PR with what you see as the issue there. It seems straightforward to me.

File: openlibrary/catalog/author/merge.py [hunk 1]
```diff
@@ -129,7 +129,7 @@ def do_normalize(author_key, best_key, authors):
             need_update = True
     else:
         best = ol.get(best_key)
-        author_keys = set(k for k in a.keys() + best.keys() if k not in ('key', 'last_modified', 'type', 'id', 'revision'))
+        author_keys = set(k for k in list(a.keys()) + list(best.keys()) if k not in ('key', 'last_modified', 'type', 'id', 'revision'))
```
Comment (cclauss) [inline openlibrary/catalog/author/merge.py]: `.keys()` are not required.  The following are all equivalent.
```
for key in a:  # unless a could change during iteration
for key in list(a):
for key in list(a.keys()):
```
In this case, `list(a) + list(best)` would work just fine.

File: openlibrary/catalog/author/rename.py [hunk 2]
```diff
@@ -120,14 +120,14 @@ def make_redirect(old, new):
         'location': {'connect': 'update', 'value': new['key'] },
         'type': {'connect': 'update', 'value': '/type/redirect' },
     }
-    for k in old.iterkeys():
+    for k in old.keys():
```
Comment (cclauss) [inline openlibrary/catalog/author/rename.py]: `for k in old:`

File: openlibrary/coverstore/code.py [hunk 3]
```diff
@@ -326,7 +326,7 @@ def is_cover_in_cluster(self, coverid):
     def get_tar_filename(self, coverid, size):
         """Returns tarfile:offset:size for given coverid.
         """
-        tarindex = coverid / 10000
+        tarindex = coverid // 10000
```
Comment (cclauss) [inline openlibrary/coverstore/code.py]: tarindex, index = divmod (coverid, 10000)

**Comment by tfmorris** [comment]
I think @cclauss has cherry picked all the useful commits from this branch into individual PRs and successfully shepherded them through the long and painful review process. Thank you!

**Pull request closed without merging**